### ISSUE TITLE
Roadmap to promote the opentelemetry tracing feature to be production ready

### ISSUE THREAD

**Issue by FrankChen021** [ClickHouse/ClickHouse#49244]
The distributed tracing feature is heavily used in our environment to deal with problems in distributed environment and it has to be proven a powerful feature for a long time that helps us solve or find out many complicated problems. 
Most of the work that are implemented in our own branch has been contributed back to the community.

The main features in the master branch now are as follows:
1. Trace on cluster DDLs
2. Trace a query on local node and its sub-queries on remote node
3. Trace async or sync INSERT on distributed table
4. Trace queries from HTTP/TCP/GRPC
5. Propagate tracing context to downstream servers via URL engine

But the status of this feature is still marked as [experienmental](https://clickhouse.com/docs/en/operations/opentelemetry). 
From the community perspective, I think it's time for us to give a plan to promot it as a production ready feature. 

Before that, here some things that I can come across to be completed:

1. Standardize the attribute names as defined in the [opentelemetry specification](https://opentelemetry.io/docs/reference/specification/trace/semantic_conventions/). 
    The attribute names now can be defined in anyway, it's better to use current specification to standardize some of them to allow the logs can be easily handled by some other external visulization tools. **This is NOT backward compatible.**
2. Investigate the root cause of https://github.com/ClickHouse/ClickHouse/issues/49185
     Even though it occurs in Debug build, but it indicates that this may lead to incorrect logs in the Release build if it happens
3. Add `trace_id` column to `system.query_log`
    This will give a clear info in the query log that if a query is traced or not. And then it can be used to search/join the `opentelemetry_span_logs` table
4. Support Materialized View in the distributed tracing
     See: https://github.com/ClickHouse/ClickHouse/pull/41672
5. Propagate the tracing context to remote S3
    Some S3-compatible remote storages support this distributed tracing feature. It would give us the ability to deal with problems between ClickHouse and underlying S3 storage.

What do you think? @alexey-milovidov

**Comment by alexey-milovidov** [comment]
1. Yes, let's change the attribute names to match some standards until it is too late.
2. This error is related to the usage of fibers due to enabling hedged requests. @Avogar is investigating it.
3. Let's do it.
4. This pull request is still a draft, let's go ahead.
5. This is of low priority, but why not... Also check https://github.com/ClickHouse/ClickHouse/pull/34585, you can continue in another PR.

**Comment by mzealey** [comment]
I've just been playing with this via python asynch library which includes some support for it. There is what seems to be non-standard behaviour with `traceparent` in that the CH `opentelemetry_span_log` takes the span id as `parent_span_id` but does not take the `trace_id` (which in python at least is not a uuid). I think that what seems to be standard behaviour in otel ecosystem is that prior to going to production-ready status (and while these other changes are being made) it should probably be changed so that CH `trace_id` can be anything, and that it accepts this from the `traceparent` parameter?

**Comment by FrankChen021** [comment]
> the CH opentelemetry_span_log takes the span id as parent_span_id but does not take the trace_id (which in python at least is not a uuid). 

If you look at the code in ClickHouse, it follows the opentelemtry standard to parse the `traceparent`. I don't know how it happens, but you can give an example.

**Comment by mzealey** [comment]
Sure, but it seems like what the behaviour of other otel systems (eg python and java integrations) is is that trace-id is also inherited from traceparent, whereas with clickhouse it is not and there is a uuid type restriction on it which would need lifting for it to be allowed

**Comment by alexey-milovidov** [comment]
Depends on #52584

**Comment by maruthgoyal** [comment]
👋  @alexey-milovidov ooc, is there an update with regards to this issue? (since the dependent issues seem to be marked clsoed)

**Comment by alexey-milovidov** [comment]
If @FrankChen021 says it is production-ready, we can remove the experimental flag.

**Comment by FrankChen021** [comment]
I don't see there will be breaking changes of this feature in future. 
Even some of the work in this issue have not been done, we can see them as improvements of this future.  
So, let's mark it as production ready.
